Framework/DataHandling/inc/MantidDataHandling/ISISJournal.h

```diff
@@ -0,0 +1,44 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidKernel/System.h"
+
+#include <map>
+#include <string>
+#include <vector>
+
+namespace Mantid {
+namespace DataHandling {
+namespace ISISJournal {
+/**
+  Defines functions to aid in fetching ISIS specific run information from
+  journal files
+ */
+
+using ISISJournalTags = std::vector<std::string>;
+using ISISJournalFilters = std::map<std::string, std::string>;
+using ISISJournalData = std::map<std::string, std::string>;
+
+std::vector<ISISJournalData> DLLExport
+getRunDataFromFile(std::string const &fileContents,
```

Comment:
Good call on the name.

This function is mainly for testing - the main entry point is to pass the instrument and cycle because it will be fetched from the web. I'll add a comment to that effect. At the moment I don't envisage needing to open any other kind of file path.

Reply:
I'm thinking actually this would be neater if I made it a class, say `ISISJournal(instrument, cycle)` and with methods `getRuns(tags, filters)` and `getCycleNames()`. Then I'll handle testing with a friend class that can set the file content from a string and remove this from the public interface.

It could then also support multiple calls to `getRuns` with different tags/filters without having to reload the file.